### PR TITLE
Automate entire release process

### PR DESCRIPTION
**Pull request by andschwa**
This PR sets up the final parts necessary for the Azure DevOps release pipeline to create a (draft) GitHub Release with the built assets directly uploaded from the pipeline itself. This eliminates any developer machine touching the bits between build, sign, and release. There is a manual approval step in the deployment such that the draft release is not created until a Microsoft maintainer approves it. It also includes a few cleanups (such as moving from "build artifacts" to "pipeline artifacts"). It was tested up to the point that the GitHub API complained the release branch didn't exist, which since I wasn't testing off a release branch, was accurate.

### CODE REVIEW

**Comment by andschwa** [comment]
I've temporarily restored the `release/v2.4.6` for one more test to see if the draft release is created and the artifact is uploaded 🤞

**Comment by andschwa** [comment]
Yay! The release it worked. @rjmholt check it: https://github.com/PowerShell/PowerShellEditorServices/releases/tag/untagged-9fb7e73f0f1ec8c30383

**Comment by andschwa** [comment]
Oops, I have to fix something.

**Comment by andschwa** [comment]
Just double-checking those new changes didn't break the release build...

**Comment by andschwa** [comment]
Ok, that's it! It all works.

**Review by rjmholt — APPROVED** [review]

File: .vsts-ci/azure-pipelines-release.yml [hunk 1]
```diff
@@ -18,24 +20,45 @@ resources:
...
+    type: git
+    name: vscode-powershell
+
+stages:
+- stage: Build
+  displayName: Build the release
+  jobs:
+  - job: Build
+    pool:
+      vmImage: vs2017-win2016
```
[the remark this answers is not in the document]
Comment (andschwa) [reply .vsts-ci/azure-pipelines-release.yml:35]: There is, actually. Another time.

**Pull request merged**